pvlib/spectrum/mismatch.py

```diff
@@ -1100,3 +1101,95 @@ def qe_to_sr(qe, wavelength=None, normalize=False):
         spectral_responsivity = normalize_max2one(spectral_responsivity)
 
     return spectral_responsivity
+
+
+def average_photon_energy(spectral_irr):
+    r"""
+    Calculate the average photon energy of one or more spectral irradiance
+    distributions.
+
+    Parameters
+    ----------
+    spectral_irr : pandas.Series or pandas.DataFrame
+
+        Spectral irradiance, must be positive. [Wm⁻²nm⁻¹]
+
+        A single spectrum must be a :py:class:`pandas.Series` with wavelength
+        [nm] as the index, while multiple spectra must be a
+        :py:class:`pandas.DataFrame` with column headers as wavelength [nm].
+
+    Returns
+    -------
+    ape : numeric or array
+        Average Photon Energy [eV].
```

Comment:
Can someone with more familiarity with the relevant literature comment on whether eV is the best/most idiomatic unit to use here?  The alternative I see is J, which maybe fits better with the rest of pvlib in some sense, but of course would result in rather unwieldy numeric values here.

Reply:
At least in the context of PV performance modelling, I have only ever seen the APE quoted in eV. I think eV rather than J is the best choice here. I will leave this comment open for others to weigh in though.